internal/pkg/logging/workload.go

```diff
@@ -173,7 +173,9 @@ func (s *WorkloadClient) WriteLogEvents(opts WriteLogEventsOpts) error {
 		LogStreamLimit:      opts.LogStreamLimit,
 	}
 
-	logEventsOpts.LogStreamPrefixFilters = s.logStreams(opts.TaskIDs)
+	if s.logStreamNamePrefix != "" {
+		logEventsOpts.LogStreamPrefixFilters = s.logStreams(opts.TaskIDs)
+	}
```

Comment:
I think our mistake is that we are returning a `*WorkloadClient` for both apprunner and ecs causing the type to have a mix of fields that don't make sense for one service type 💭 .